### ISSUE TITLE
Cannot load templates - 5.4.3

### ISSUE THREAD

**Issue by Danbown** [Blazemeter/CorrelationRecorder#31]
Hi, I added the Correlation Recorder through plugin manager, using JMeter 5.4.3 on Windows 10.
 It is possible to add a new Group and create new rules and save but I am unable to load any templates.
Eg when I clear templates and attempt to reload with 'Load Template', nothing comes up. 
Also, although I can save a template, when I try to click save again, the saving template appears and requires all information to be retyped.

Error logs are at bottom of post.

**Environment: 
Windows 10
Networked device behind corporate proxy
JMeter version 5.4.3
bzm-correlation-recorder v 1.2.2 updated today

Steps to reproduce the bug:

1. Open JMeter through command line with parameters to work with corporate proxy.
2. Open existing jmx file or create new plan from template
3. Navigate to bzm - Correlation Recorder
4. Click 'Correlation Rules' tab
5. Click 'Load Template'

Observe that nothing happens
Observe that new entry appears in error logs, see bottom of this post
Note that json file is saved here: C:\Work\JMeter\apache-jmeter-5.4.3\correlation-templates\DanB_Template-1.0-template.json

**Expected behavior**

Once saved, it should be possible to load templates in existing or new test plans.

**Complementary information**

C:\Work\JMeter\apache-jmeter-5.4.3\correlation-templates\DanB_Template-1.0-template.json contains:
```json
{
  "id" : "DanB_Template",
  "description" : "First draft of template for PoC",
  "version" : "1.0",
  "author" : "DanB",
  "url" : "",
  "components" : "",
  "responseFilters" : "",
  "groups" : [ {
    "id" : "Group-1",
    "rules" : [ {
      "referenceName" : "state",
      "correlationExtractor" : {
        "type" : "com.blazemeter.jmeter.correlation.core.extractors.RegexCorrelationExtractor",
        "multiValued" : true,
        "regex" : "state=(.*?)\\\\"
      },
      "correlationReplacement" : {
        "type" : "com.blazemeter.jmeter.correlation.core.replacements.RegexCorrelationReplacement",
        "regex" : "state=(.*?)\\\\"
      },
      "enabled" : true
    }, {
      "referenceName" : "nonce",
      "correlationExtractor" : {
        "type" : "com.blazemeter.jmeter.correlation.core.extractors.RegexCorrelationExtractor",
        "multiValued" : true,
        "regex" : "nonce=(.*?)\\\\"
      },
      "correlationReplacement" : {
        "type" : "com.blazemeter.jmeter.correlation.core.replacements.RegexCorrelationReplacement",
        "regex" : "nonce=(.*?)\\\\\""
      },
      "enabled" : true
    } ],
    "enable" : true
  } ],
  "dependencies" : [ ],
  "repositoryId" : "local",
  "changes" : "first draft",
  "installed" : false
}
```
**Additional context**
```java
2022-12-07 14:51:34,430 WARN c.b.j.c.c.t.LocalCorrelationTemplatesRepositoriesRegistry: There was an issue trying to read the file central-repository.json.
com.fasterxml.jackson.databind.exc.MismatchedInputException: No content to map due to end-of-input
 at [Source: (File); line: 1, column: 0]
	at com.fasterxml.jackson.databind.exc.MismatchedInputException.from(MismatchedInputException.java:59) ~[jackson-databind-2.12.0.jar:2.12.0]
	at com.fasterxml.jackson.databind.ObjectMapper._initForReading(ObjectMapper.java:4686) ~[jackson-databind-2.12.0.jar:2.12.0]
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4584) ~[jackson-databind-2.12.0.jar:2.12.0]
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3431) ~[jackson-databind-2.12.0.jar:2.12.0]
	at com.blazemeter.jmeter.correlation.core.templates.LocalConfiguration.readTemplatesReferences(LocalConfiguration.java:132) ~[jmeter-bzm-correlation-recorder-1.2.2.jar:?]
	at com.blazemeter.jmeter.correlation.core.templates.LocalCorrelationTemplatesRepositoriesRegistry.readTemplatesReferences(LocalCorrelationTemplatesRepositoriesRegistry.java:124) ~[jmeter-bzm-correlation-recorder-1.2.2.jar:?]
	at com.blazemeter.jmeter.correlation.core.templates.LocalCorrelationTemplatesRepositoriesRegistry.lambda$getRepositories$0(LocalCorrelationTemplatesRepositoriesRegistry.java:111) ~[jmeter-bzm-correlation-recorder-1.2.2.jar:?]
	at java.util.ArrayList.forEach(ArrayList.java:1511) ~[?:?]
	at com.blazemeter.jmeter.correlation.core.templates.LocalCorrelationTemplatesRepositoriesRegistry.getRepositories(LocalCorrelationTemplatesRepositoriesRegistry.java:100) ~[jmeter-bzm-correlation-recorder-1.2.2.jar:?]
	at com.blazemeter.jmeter.correlation.core.templates.CorrelationTemplatesRepositoriesConfiguration.getCorrelationRepositories(CorrelationTemplatesRepositoriesConfiguration.java:51) ~[jmeter-bzm-correlation-recorder-1.2.2.jar:?]
	at com.blazemeter.jmeter.correlation.CorrelationProxyControl.getCorrelationRepositories(CorrelationProxyControl.java:613) ~[jmeter-bzm-correlation-recorder-1.2.2.jar:?]
	at com.blazemeter.jmeter.correlation.CorrelationProxyControlGui.getCorrelationRepositories(CorrelationProxyControlGui.java:184) ~[jmeter-bzm-correlation-recorder-1.2.2.jar:?]
	at com.blazemeter.jmeter.correlation.gui.templates.CorrelationTemplatesFrame.updateTemplatesList(CorrelationTemplatesFrame.java:321) ~[jmeter-bzm-correlation-recorder-1.2.2.jar:?]
	at com.blazemeter.jmeter.correlation.gui.templates.CorrelationTemplatesFrame.prepareInfoPanel(CorrelationTemplatesFrame.java:134) ~[jmeter-bzm-correlation-recorder-1.2.2.jar:?]
	at com.blazemeter.jmeter.correlation.gui.templates.CorrelationTemplatesFrame.<init>(CorrelationTemplatesFrame.java:125) ~[jmeter-bzm-correlation-recorder-1.2.2.jar:?]
	at com.blazemeter.jmeter.correlation.gui.RulesContainer.actionPerformed(RulesContainer.java:131) ~[jmeter-bzm-correlation-recorder-1.2.2.jar:?]
	at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1972) ~[?:?]
	at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2313) ~[?:?]
	at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:405) ~[?:?]
	at javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:262) ~[?:?]
	at javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:279) ~[?:?]
	at java.awt.Component.processMouseEvent(Component.java:6626) ~[?:?]
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3389) ~[?:?]
	at java.awt.Component.processEvent(Component.java:6391) ~[?:?]
	at java.awt.Container.processEvent(Container.java:2266) ~[?:?]
	at java.awt.Component.dispatchEventImpl(Component.java:5001) ~[?:?]
	at java.awt.Container.dispatchEventImpl(Container.java:2324) ~[?:?]
	at java.awt.Component.dispatchEvent(Component.java:4833) ~[?:?]
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4948) ~[?:?]
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4575) ~[?:?]
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4516) ~[?:?]
	at java.awt.Container.dispatchEventImpl(Container.java:2310) ~[?:?]
	at java.awt.Window.dispatchEventImpl(Window.java:2780) ~[?:?]
	at java.awt.Component.dispatchEvent(Component.java:4833) ~[?:?]
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:773) ~[?:?]
	at java.awt.EventQueue$4.run(EventQueue.java:722) ~[?:?]
	at java.awt.EventQueue$4.run(EventQueue.java:716) ~[?:?]
	at java.security.AccessController.doPrivileged(AccessController.java:399) [?:?]
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86) ~[?:?]
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:97) ~[?:?]
	at java.awt.EventQueue$5.run(EventQueue.java:746) ~[?:?]
	at java.awt.EventQueue$5.run(EventQueue.java:744) ~[?:?]
	at java.security.AccessController.doPrivileged(AccessController.java:399) [?:?]
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86) [?:?]
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:743) [?:?]
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203) [?:?]
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124) [?:?]
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113) [?:?]
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109) [?:?]
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101) [?:?]
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:90) [?:?]
2022-12-07 14:51:34,451 ERROR o.a.j.JMeter: Uncaught exception in thread Thread[AWT-EventQueue-0,6,main]
java.lang.reflect.InaccessibleObjectException: Unable to make field private int java.awt.image.BufferedImage.imageType accessible: module java.desktop does not "opens java.awt.image" to unnamed module @501edcf1
	at java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:354) ~[?:?]
	at java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297) ~[?:?]
	at java.lang.reflect.Field.checkCanSetAccessible(Field.java:178) ~[?:?]
	at java.lang.reflect.Field.setAccessible(Field.java:172) ~[?:?]
	at com.fasterxml.jackson.databind.util.ClassUtil.checkAndFixAccess(ClassUtil.java:987) ~[jackson-databind-2.12.0.jar:2.12.0]
	at com.fasterxml.jackson.databind.deser.impl.FieldProperty.fixAccess(FieldProperty.java:104) ~[jackson-databind-2.12.0.jar:2.12.0]
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBuilder._fixAccess(BeanDeserializerBuilder.java:522) ~[jackson-databind-2.12.0.jar:2.12.0]
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBuilder.build(BeanDeserializerBuilder.java:373) ~[jackson-databind-2.12.0.jar:2.12.0]
	at com.fasterxml.jackson.databind.deser.BeanDeserializerFactory.buildBeanDeserializer(BeanDeserializerFactory.java:293) ~[jackson-databind-2.12.0.jar:2.12.0]
	at com.fasterxml.jackson.databind.deser.BeanDeserializerFactory.createBeanDeserializer(BeanDeserializerFactory.java:150) ~[jackson-databind-2.12.0.jar:2.12.0]
	at com.fasterxml.jackson.databind.deser.DeserializerCache._createDeserializer2(DeserializerCache.java:414) ~[jackson-databind-2.12.0.jar:2.12.0]
	at com.fasterxml.jackson.databind.deser.DeserializerCache._createDeserializer(DeserializerCache.java:349) ~[jackson-databind-2.12.0.jar:2.12.0]
	at com.fasterxml.jackson.databind.deser.DeserializerCache._createAndCache2(DeserializerCache.java:264) ~[jackson-databind-2.12.0.jar:2.12.0]
	at com.fasterxml.jackson.databind.deser.DeserializerCache._createAndCacheValueDeserializer(DeserializerCache.java:244) ~[jackson-databind-2.12.0.jar:2.12.0]
	at com.fasterxml.jackson.databind.deser.DeserializerCache.findValueDeserializer(DeserializerCache.java:142) ~[jackson-databind-2.12.0.jar:2.12.0]
	at com.fasterxml.jackson.databind.DeserializationContext.findNonContextualValueDeserializer(DeserializationContext.java:581) ~[jackson-databind-2.12.0.jar:2.12.0]
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.resolve(BeanDeserializerBase.java:524) ~[jackson-databind-2.12.0.jar:2.12.0]
	at com.fasterxml.jackson.databind.deser.DeserializerCache._createAndCache2(DeserializerCache.java:293) ~[jackson-databind-2.12.0.jar:2.12.0]
	at com.fasterxml.jackson.databind.deser.DeserializerCache._createAndCacheValueDeserializer(DeserializerCache.java:244) ~[jackson-databind-2.12.0.jar:2.12.0]
	at com.fasterxml.jackson.databind.deser.DeserializerCache.findValueDeserializer(DeserializerCache.java:142) ~[jackson-databind-2.12.0.jar:2.12.0]
	at com.fasterxml.jackson.databind.DeserializationContext.findRootValueDeserializer(DeserializationContext.java:591) ~[jackson-databind-2.12.0.jar:2.12.0]
	at com.fasterxml.jackson.databind.ObjectMapper._findRootDeserializer(ObjectMapper.java:4731) ~[jackson-databind-2.12.0.jar:2.12.0]
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4592) ~[jackson-databind-2.12.0.jar:2.12.0]
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3411) ~[jackson-databind-2.12.0.jar:2.12.0]
	at com.blazemeter.jmeter.correlation.core.templates.LocalConfiguration.readValue(LocalConfiguration.java:127) ~[jmeter-bzm-correlation-recorder-1.2.2.jar:?]
	at com.blazemeter.jmeter.correlation.core.templates.LocalCorrelationTemplatesRepositoriesRegistry.lambda$getCorrelationTemplatesByRepositoryId$3(LocalCorrelationTemplatesRepositoriesRegistry.java:190) ~[jmeter-bzm-correlation-recorder-1.2.2.jar:?]
	at java.util.ArrayList.forEach(ArrayList.java:1511) ~[?:?]
	at com.blazemeter.jmeter.correlation.core.templates.LocalCorrelationTemplatesRepositoriesRegistry.getCorrelationTemplatesByRepositoryId(LocalCorrelationTemplatesRepositoriesRegistry.java:188) ~[jmeter-bzm-correlation-recorder-1.2.2.jar:?]
	at com.blazemeter.jmeter.correlation.core.templates.CorrelationTemplatesRepositoriesConfiguration.getCorrelationTemplatesByRepositoryName(CorrelationTemplatesRepositoriesConfiguration.java:55) ~[jmeter-bzm-correlation-recorder-1.2.2.jar:?]
	at com.blazemeter.jmeter.correlation.CorrelationProxyControl.getCorrelationTemplatesByRepositoryName(CorrelationProxyControl.java:624) ~[jmeter-bzm-correlation-recorder-1.2.2.jar:?]
	at com.blazemeter.jmeter.correlation.CorrelationProxyControlGui.getCorrelationTemplatesByRepositoryName(CorrelationProxyControlGui.java:194) ~[jmeter-bzm-correlation-recorder-1.2.2.jar:?]
	at com.blazemeter.jmeter.correlation.gui.templates.CorrelationTemplatesFrame.lambda$updateTemplatesList$3(CorrelationTemplatesFrame.java:323) ~[jmeter-bzm-correlation-recorder-1.2.2.jar:?]
	at java.util.ArrayList.forEach(ArrayList.java:1511) ~[?:?]
	at com.blazemeter.jmeter.correlation.gui.templates.CorrelationTemplatesFrame.updateTemplatesList(CorrelationTemplatesFrame.java:321) ~[jmeter-bzm-correlation-recorder-1.2.2.jar:?]
	at com.blazemeter.jmeter.correlation.gui.templates.CorrelationTemplatesFrame.prepareInfoPanel(CorrelationTemplatesFrame.java:134) ~[jmeter-bzm-correlation-recorder-1.2.2.jar:?]
	at com.blazemeter.jmeter.correlation.gui.templates.CorrelationTemplatesFrame.<init>(CorrelationTemplatesFrame.java:125) ~[jmeter-bzm-correlation-recorder-1.2.2.jar:?]
	at com.blazemeter.jmeter.correlation.gui.RulesContainer.actionPerformed(RulesContainer.java:131) ~[jmeter-bzm-correlation-recorder-1.2.2.jar:?]
	at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1972) ~[?:?]
	at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2313) ~[?:?]
	at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:405) ~[?:?]
	at javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:262) ~[?:?]
	at javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:279) ~[?:?]
	at java.awt.Component.processMouseEvent(Component.java:6626) ~[?:?]
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3389) ~[?:?]
	at java.awt.Component.processEvent(Component.java:6391) ~[?:?]
	at java.awt.Container.processEvent(Container.java:2266) ~[?:?]
	at java.awt.Component.dispatchEventImpl(Component.java:5001) ~[?:?]
	at java.awt.Container.dispatchEventImpl(Container.java:2324) ~[?:?]
	at java.awt.Component.dispatchEvent(Component.java:4833) ~[?:?]
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4948) ~[?:?]
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4575) ~[?:?]
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4516) ~[?:?]
	at java.awt.Container.dispatchEventImpl(Container.java:2310) ~[?:?]
	at java.awt.Window.dispatchEventImpl(Window.java:2780) ~[?:?]
	at java.awt.Component.dispatchEvent(Component.java:4833) ~[?:?]
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:773) ~[?:?]
	at java.awt.EventQueue$4.run(EventQueue.java:722) ~[?:?]
	at java.awt.EventQueue$4.run(EventQueue.java:716) ~[?:?]
	at java.security.AccessController.doPrivileged(AccessController.java:399) ~[?:?]
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86) ~[?:?]
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:97) ~[?:?]
	at java.awt.EventQueue$5.run(EventQueue.java:746) ~[?:?]
	at java.awt.EventQueue$5.run(EventQueue.java:744) ~[?:?]
	at java.security.AccessController.doPrivileged(AccessController.java:399) ~[?:?]
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86) ~[?:?]
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:743) ~[?:?]
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203) [?:?]
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124) [?:?]
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113) [?:?]
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109) [?:?]
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101) [?:?]
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:90) [?:?]
```

**Comment by RicardoPoleo** [comment]
Hello @Danbown,

Thanks for taking the time to both provide all the info and report this issue.

I tested the JSON in a JSON linter (like https://jsonlint.com/) and, apparently, there is an extra `"`at the end of the following regular expression (which is breaking the parsing).
`"regex" : "nonce=(.*?)\\\\\""`

I'm assuming you want your regular expression to have a double quote on it? If that's the case, I would recommend using only one escape, we did something similar for the WordPress template on the Central repository:
`"regex" : "_wpnonce=(.*)\""`

Could you try reviewing the JSON and load it again, those escapes sometimes can be confusing 😉.

**Comment by Danbown** [comment]
Hi @RicardoPoleo 
thanks for your prompt reply :)
I fixed the JSON file as above, directly in VS code but am still unable to load a Template - nothing happens in the UI when I click the button but I got the same entries in the error log as before, even after fixing my new template (compared the two error entries in Notepad++ and only the timestamps differ).

The first part of the error message reads (full version posted earlier):
"2022-12-08 00:13:15,138 WARN c.b.j.c.c.t.LocalCorrelationTemplatesRepositoriesRegistry: There was an issue trying to read the file central-repository.json.
com.fasterxml.jackson.databind.exc.MismatchedInputException: No content to map due to end-of-input
 at [Source: (File); line: 1, column: 0]"

My file 'central-repository.json' is saved in a subfolder of the 'correlation-templates' folder. The file itself is blank/0 KB. Possibly because GitHub is blocked by firewall settings? If this is important, I cannot get around this block...
Also, not sure whether this is relevant but under 'Additional Correlation Extractors', there is only one option - the 'Siebel Row Correlation Extractor' (shown under 'Available'). Unsure if Regex is supposed to be listed or whether this is implicit (since it is obviously present).
Many thanks Dan

**Comment by 3dgiordano** [comment]
Hi @Danbown  

We made a release with several compatibility improvements.
Could you carry out tests with this version and tell us if it solves the problems reported?

https://github.com/Blazemeter/CorrelationRecorder/releases/tag/v1.2.3
